module/synchronization/metrics.go

```diff
@@ -0,0 +1,138 @@
+
+	RangeRequested(ran flow.Range)
+
+	BatchRequested(batch flow.Batch)
+}
+
+type NoopMetrics struct{}
+
+func (nc *NoopMetrics) PrunedBlockById(status *Status)                                        {}
+func (nc *NoopMetrics) PrunedBlockByHeight(status *Status)                                    {}
+func (nc *NoopMetrics) PrunedBlocks(totalByHeight, totalById, storedByHeight, storedById int) {}
+func (nc *NoopMetrics) RangeRequested(ran flow.Range)                                         {}
+func (nc *NoopMetrics) BatchRequested(batch flow.Batch)                                       {}
+
+const (
+	namespaceSynchronization = "synchronization"
+	subsystemSyncCore        = "sync_core"
+)
+
+type MetricsCollector struct {
+	timeToPruned          *prometheus.HistogramVec
+	timeToReceived        *prometheus.HistogramVec
+	totalPruned           *prometheus.CounterVec
+	storedByHeight        prometheus.Gauge
+	storedById            prometheus.Gauge
+	totalHeightsRequested prometheus.Counter
+	totalIdsRequested     prometheus.Counter
+}
+
+func NewMetricsCollector() *MetricsCollector {
+	return &MetricsCollector{
+		timeToPruned: prometheus.NewHistogramVec(prometheus.HistogramOpts{
+			Name:      "time_to_pruned",
+			Namespace: namespaceSynchronization,
+			Subsystem: subsystemSyncCore,
+			Help:      "the time between queueing and pruning a block in milliseconds",
+			Buckets:   []float64{100, 250, 500, 1000, 2500, 5000, 7500, 10000, 20000},
+		}, []string{"status", "requested_by"}),
+		timeToReceived: prometheus.NewHistogramVec(prometheus.HistogramOpts{
+			Name:      "time_to_received",
+			Namespace: namespaceSynchronization,
+			Subsystem: subsystemSyncCore,
+			Help:      "the time between queueing and receiving a block in milliseconds",
+			Buckets:   []float64{100, 250, 500, 1000, 2500, 5000, 7500, 10000, 20000},
+		}, []string{"requested_by"}),
+		totalPruned: prometheus.NewCounterVec(prometheus.CounterOpts{
+			Name:      "total_blocks_pruned",
+			Namespace: namespaceSynchronization,
+			Subsystem: subsystemSyncCore,
+			Help:      "the total number of blocks pruned by 'id' or 'height'",
+		}, []string{"requested_by"}),
+		storedByHeight: prometheus.NewGauge(prometheus.GaugeOpts{
+			Name:      "blocks_stored_by_height",
+			Namespace: namespaceSynchronization,
+			Subsystem: subsystemSyncCore,
+			Help:      "the number of blocks currently stored by height",
+		}),
+		storedById: prometheus.NewGauge(prometheus.GaugeOpts{
+			Name:      "blocks_stored_by_id",
+			Namespace: namespaceSynchronization,
+			Subsystem: subsystemSyncCore,
+			Help:      "the number of blocks currently stored by id",
+		}),
+		totalHeightsRequested: prometheus.NewCounter(prometheus.CounterOpts{
+			Name:      "total_heights_requested",
+			Namespace: namespaceSynchronization,
+			Subsystem: subsystemSyncCore,
+			Help:      "the total number of blocks requested by height. range of 0-10 would increase the counter by 10",
```

Comment:
```suggestion
			Help:      "the total number of blocks requested by height, including retried requests for the same heights. Eg: a range of 1-10 would increase the counter by 10",
```
Adding a note that this includes retries, and fixing an off-by-one (ranges are inclusive on both boundaries)